index.d.ts

```diff
@@ -0,0 +1,77 @@
+export interface JsonMLAttributes {
+  [key: string]: any // @TODO - can this be anything but a string? it technically SHOULD be,
+      // but I'm not sure if we want to be that strict. maybe it should be primitives at least?
+      // or maybe we just leave it alone?
+}
```

Comment:
1. I'm unsure about naming in general - whether to be verbose like I've done here, or to be simple like `Attributes` and `Node` or `Element` etc
2. attribute values are a bit tricky. Anything parsed from an HTML will be in string format, always. Any JsonML attributes converted back into HTML or DOM are coerced to strings. But does that mean that, in the interim, they must be strings? Also, I know there are plenty of use cases where just pragmatically the values have been modified and altered and are definitely NOT strings. I think I'm in favor of leaving this  set to `any` rather than imposing strictness.

Reply:
1. I've gone w/ the generic naming.
2. For values, I've defined a type to centralize the issue but I'm still debating a bit. I suspect our application usage (appending things to JsonML structures) will potentially conflict with the strict value here. Pragmatically, I worry this could cause some problems for us in practice and I'm wary of implementing it. _In theory_, any value could be ok if it as a `toString()` implementation anyway. This library could export both types for consuming applications to optionally make use of? 🤷‍♂

Reply:
2. I'm suggesting, for example:
```
type Attribute = string | number | <other primitive types here>;
```
The problem I see with using `any` is you open the doors to object literals and on and on, things which will not, to your point, have a `toString()` implementation.

Reply:
`question`: Ah...I see you have `AttributeValueStrict` now. Why are you not using it?

Reply:
I'm not using it because I think it will break our applications! I'm _pretty sure_ we are extending element attributes with objects within our applications, since we don't intend to transform the JsonML back into HTML or DOM. An example of a common use case might be an embedded content reference where later we merge in the related data structures directly with the attributes. Enforcing this type strictness is correct per the spec but might become a problem for us.

Reply:
I gave this some more thought. This fork is open-source and shouldn't care about how downstream applications. The type definitions should be strict and accurate. If we need to find some workarounds later, we can figure that out.

Reply:
In my experience, its pretty common to have strict types imported as correctly typed by a lib and then extend, union, or partial them in my code. Here's a paired down example of how this might work in this case: [Playground Link](https://www.typescriptlang.org/play/#code/C4TwDgpgBAgswCcCWAjArsCA1AhgGzWgF4oBnRJAOwHMoAfKStAWxQgXqhQHtu8IclTkzx4A3ACgAxt0rkoAN3yEAXLHjJ0mXAWJQA3gF9JE0JCgBZEHApbsyvTc0Z7uztxQArCFOCSZcsBQzCA6qpbWGqguYXpGkkA).

Reply:
Right, agreed. The trouble is that the `AttributeValue` will be used as part of other type definitions in this file. That's where it all goes sideways.

Reply:
@jeromecovington @pgoldrbx yeah, I think for now we may just want to use `any` for pragmatic reasons since downstream consumers haven't adhered to the spec when performing manual transformations of the JsonML after the initial document conversion to JsonML from Markdown. If we're wanting to strictly adhere to the spec, we may want to do this in a two-phase approach, where the first is to publish a version using `any` here and then issue a breaking change that follows the spec, giving consumers a chance to adjust approach on those not-to-spec implementations